inginious/frontend/plugins/upcoming_tasks/__init__.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+#
+
+""" A plugin that allow students to see their futur work in a single place for all their courses """
+import os
+from collections import OrderedDict
+from datetime import datetime, timedelta
+import flask
+
+
+from flask import send_from_directory
+from inginious.frontend.pages.utils import INGIniousPage, INGIniousAuthPage
+
+PATH_TO_PLUGIN = os.path.abspath(os.path.dirname(__file__))
+
+def menu(template_helper):
+    """ Displays the link to the board on the main page, if the plugin is activated """
+    return template_helper.render("main_menu.html", template_folder=PATH_TO_PLUGIN + '/templates/')
+
+class StaticMockPage(INGIniousPage):
+    """ MockPage based on auto-evaluation plugin structure """
+
+    def GET(self, path):
+        return send_from_directory(os.path.join(PATH_TO_PLUGIN, "static"), path)
+
+    def POST(self, path):
+        return self.GET(path)
+
+
+class UpComingTasksBoard(INGIniousAuthPage):
+
+    def GET_AUTH(self):
+        """ Called when reaching the page """
+        time_planner = "unlimited"
+        return self.page(time_planner)
+
+    def POST_AUTH(self):
+        """ Called when modifying time planner """
+        user_input = flask.request.form
+        time_planner = "unlimited"
+        if "time_planner" in user_input:
+            time_planner = user_input.get("time_planner")
+        return self.page(time_planner)
+
+    def time_planner_conversion(self, string_time_planner):
+        """ Used to convert the time_planner options into int value """
+        if string_time_planner == "unlimited":
+            return 100000
+        return int(string_time_planner)
```

Comment:
Here is the related form:
![image](https://user-images.githubusercontent.com/17684696/115682178-82fb7f80-a355-11eb-9725-f0cba76d6298.png)

So the value is always "7", "14", "30" or "unlimited".
In other words, there is 100% no bug here in the current version but indeed if it was to be adapted or to allow any value then it might require some checks.

Do you want me to change this or should I let it be like this ?

Reply:
I can perform a post request that will raise an exception.

Reply:
Well, clearly I didn't take into account manual requests.
Right now, they would raise an exception how do you suggest to handle this?
I thought it they raised an exception it would actually be a good thing

Reply:
Is this better ?    

def time_planner_conversion(self, string_time_planner):
        """ Used to convert the time_planner options into int value """
        if string_time_planner in ["7", "14", "30"]:
            return int(string_time_planner)
        return 100000